src/Storage/Device/S3.php

```
@@ -415,10 +432,14 @@ private function getInfo(string $path)
             $rest->error = ['code' => $rest->code, 'message' => 'Unexpected HTTP status'];
```

We can just throw an exception here.